components/cli/pkg/commands/list_ingresses.go

```diff
@@ -200,110 +202,164 @@ func displayImageApisTable(cellImageName string) {
+			}
+			if ingressInfo.IngressTypeTCP == constants.TCP_INGRESS {
+				ingressData = append(ingressData, fmt.Sprintf("%s_%s, %s_tcp_%s", componentDetail.ComponentName,
+					constants.HOST, componentDetail.ComponentName, constants.PORT))
+			} else {
+				ingressData = append(ingressData, fmt.Sprintf("%s_%s, %s_%s", componentDetail.ComponentName,
+					constants.HOST, componentDetail.ComponentName, constants.PORT))
+			}
+			tableData = append(tableData, ingressData)
 		}
 	}
+
 	table := tablewriter.NewWriter(os.Stdout)
-	table.SetHeader([]string{"COMPONENT", "INGRESS TYPE", "INGRESS PORT"})
+	table.SetHeader([]string{"COMPONENT", "INGRESS TYPE", "INGRESS PORT", "INGRESS_KEY"})
 	table.SetBorders(tablewriter.Border{Left: false, Top: false, Right: false, Bottom: false})
 	table.SetAlignment(3)
 	table.SetRowSeparator("-")
 	table.SetCenterSeparator(" ")
 	table.SetColumnSeparator(" ")
 	table.SetHeaderColor(
+		tablewriter.Colors{tablewriter.Bold},
 		tablewriter.Colors{tablewriter.Bold},
 		tablewriter.Colors{tablewriter.Bold},
 		tablewriter.Colors{tablewriter.Bold})
 	table.SetColumnColor(
+		tablewriter.Colors{},
 		tablewriter.Colors{},
 		tablewriter.Colors{},
 		tablewriter.Colors{})
 	table.AppendBulk(tableData)
 	table.Render()
 }
 
-func displayCellImageApisTable(cellImageContent *image.Cell) {
+func displayCellImageApisTable(cellImageContent string) {
+	cell := getIngressValues(cellImageContent)
 	var tableData [][]string
-	for _, component := range cellImageContent.Spec.Components {
-		componentName := component.Metadata.Name
-		// Iterate HTTP and Web ingresses
-		for _, ingress := range cellImageContent.Spec.Gateway.Spec.Ingress.HTTP {
-			backend := ingress.Destination.Host
-			if componentName == backend {
-				var ingressData []string
-				ingressData = append(ingressData, componentName)
-
-				var ingressType = "web"
-				if len(cellImageContent.Spec.Gateway.Spec.Ingress.Extensions.ClusterIngress.Host) == 0 {
-					ingressType = "http"
-				}
-				ingressData = append(ingressData, ingressType)
-				ingressData = append(ingressData, ingress.Context)
-				ingressData = append(ingressData, ingress.Version)
-				ingressData = append(ingressData, strconv.Itoa(ingress.Port))
-				if ingressType == "web" || ingress.Global {
-					ingressData = append(ingressData, "True")
-				} else {
-					ingressData = append(ingressData, "False")
+	for _, componentDetail := range cell.Component {
+		for ingress, ingressInfo := range componentDetail.Ingress {
+			if ingressInfo.Expose == "global" {
+				ingressInfo.Expose = "true"
+			} else {
+				ingressInfo.Expose = "false"
+			}
+			if ingressInfo.IngressType == constants.HTTP_API_INGRESS && ingressInfo.Context != "" {
+				for _, resourcesValue := range ingressInfo.Definition {
+					for _, resource := range resourcesValue {
+						var ingressData []string
+						ingressData = append(ingressData, componentDetail.ComponentName)
```

Comment:
This is duplicated in multiple places, for example "ingressData = append(ingressData, componentDetail.ComponentName)" is common for all ingress types.

Reply:
ingressData []string array needs to be declared at every ingress entry. So we need to duplicate this code to have ingress details.
If we do not declare ingressData []string array , it will continue on adding data and we can not populate the table.

Reply:
IMHO ingressData []string array only needs to be declared at the beginning of the loop. Inside the loop assign values to "type", "version", "port" etc. When some field differentiates based on the ingress type, have the "if" condition and assign the value accordingly.